pages/blog/react-as-generator-engine.md

```diff
@@ -0,0 +1,218 @@
+const content = render(<FunctionComponent />);
+```
+
+Unfornatelly we have some restrictions:
+
+- React hooks is not allowed.
+- HTML tags at the moment is not supported.
+- React internal components like `Fragments`, `Suspense` etc. are skipped.
+
+## Comparison with Nunjucks
+
+If you don't know, the existing templates developers were forced (if they used our Generator) to use [Nunjucks](https://mozilla.github.io/nunjucks/). It's a templating language, heavily focused on string literals, filters (similar to bash pipes) and partials called macros.
+
+Below are some examples comparing doing those same things in Nunjucks and React. If you want check another, more complex, examples please see the updated [`template-for-generator-templates`](https://github.com/asyncapi/template-for-generator-templates) repository and the corresponding old [Nunjucks](https://github.com/asyncapi/template-for-generator-templates/tree/nunjucks) branch.
+
+### Make reusable parts
+
+It may sound trivial, but when writing any code, even a template, a programmer wants to create reusable parts that will separate the given, very frequent repeating logic.
+
+As we mentioned, in Nunjucks partials are called macros, in React reusables are called `components` (we follow with React's nomenclature). Imagine that you are writing a template that produces a Markdown content. You need to create a reusable macro/component that will be able to render a list based on given an array of strings.
+
+For example, using Nunjucks you can write something like:
+
+```njk
+{% macro list(data, type = "-") %}
+{% for item in data %}
+{{type}} {{item}}
+{% endfor %}
+{% endmacro %}
+
+{% from "partials/list.njk" import list %}
+{{ list(["one", "two", "three"]) }}
+```
+
+In React we can write it in this way:
+
+```js
+function List({ list = [], type = "-" }) {
+  return list.map(item => `${type} ${item}\n`);
+}
+
+// use `List` component in another component
+export function SimpleList() {
+  return (
+    <List list={["one", "two", "three"]} />
+  );
+}
+```
+
+Looking at both examples we see that in Nujucks we operate on string literals, it means that when passing data to a macro, you always need to know what type of data the macro takes. In React we operate on JS objects/variables. By this, your IDE should always inform you what value, of what type, you must pass to component. Also, Nunjucks's macro should be created inside `.partials` folder of template. Using React, you can create component whenever you want.
+
+### Using third party packages
+
+Using helper functions from third party packages, in Nunjucks you must apply them as [filters](https://github.com/asyncapi/generator/blob/master/docs/authoring.md#filters). For example, you want to use one function from [`Lodash`](https://lodash.com/) library like `lowerCase`. To do this, you must create a function inside `filters` folder to convert the function to Nunjucks's filter:
```

Comment:
you need to work on different example, because `lodash` is an exception here. Lodash functions are available in your templates by default if you configure generic filters library -> https://github.com/asyncapi/generator-filters

Reply:
Yes, I know that we have a separate package with lodash filters, but it's only simple case to visualize what boilerplate you must to do when you use third party packages.

Reply:
just keep in mind that users that know about this will also question this example ;) just saying`